cmd/sovereignnode/dataCodec/dataCodec_test.go

```diff
@@ -0,0 +1,452 @@
+	"github.com/stretchr/testify/require"
+)
+
+func createDataCodec() DataCodecProcessor {
+	codec := abi.NewDefaultCodec()
+	args := DataCodec{
+		Serializer: abi.NewSerializer(codec),
+		Marshaller: &marshallerMock.MarshalizerMock{},
+	}
+
+	dataCodecMock, _ := NewDataCodec(args)
+
+	return dataCodecMock
+}
+
+func TestNewDataCodec(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should work", func(t *testing.T) {
+		t.Parallel()
+
+		codec := abi.NewDefaultCodec()
+		args := DataCodec{
+			Serializer: abi.NewSerializer(codec),
+			Marshaller: &marshallerMock.MarshalizerMock{},
+		}
+		dataCodec, err := NewDataCodec(args)
+		require.Nil(t, err)
+		require.False(t, dataCodec.IsInterfaceNil())
+	})
+	t.Run("nil serializer should error", func(t *testing.T) {
+		t.Parallel()
+
+		args := DataCodec{
+			Serializer: nil,
+			Marshaller: &marshallerMock.MarshalizerMock{},
+		}
+		dataCodec, err := NewDataCodec(args)
+		require.ErrorIs(t, errors.ErrNilSerializer, err)
+		require.True(t, dataCodec.IsInterfaceNil())
+	})
+	t.Run("nil marshaller should error", func(t *testing.T) {
+		t.Parallel()
+
+		codec := abi.NewDefaultCodec()
+		args := DataCodec{
+			Serializer: abi.NewSerializer(codec),
+			Marshaller: nil,
+		}
+		dataCodec, err := NewDataCodec(args)
+		require.ErrorIs(t, errors.ErrNilMarshalizer, err)
+		require.True(t, dataCodec.IsInterfaceNil())
+	})
+}
+
+func TestDataCodec_SerializeEventData(t *testing.T) {
+	t.Parallel()
+
+	dataCodec := createDataCodec()
+
+	t.Run("nil transfer data should work", func(t *testing.T) {
+		t.Parallel()
+
+		eventData := sovereign.EventData{
+			Nonce:        10,
+			TransferData: nil,
+		}
+
+		serialized, err := dataCodec.SerializeEventData(eventData)
+		require.Nil(t, err)
+		require.Equal(t, "000000000000000a000000", hex.EncodeToString(serialized))
+	})
+	t.Run("defined transfer data should work", func(t *testing.T) {
+		t.Parallel()
+
+		eventData := sovereign.EventData{
+			Nonce: 10,
+			TransferData: &sovereign.TransferData{
+				GasLimit: 20000000,
+				Function: []byte("add"),
+				Args:     [][]byte{big.NewInt(20000000).Bytes()},
+			},
+		}
+
+		serialized, err := dataCodec.SerializeEventData(eventData)
+		require.Nil(t, err)
+		require.Equal(t, "000000000000000a010000000001312d00010000000361646401000000010000000401312d00", hex.EncodeToString(serialized))
+	})
+	t.Run("defined transfer data empty arg should work", func(t *testing.T) {
+		t.Parallel()
+
+		eventData := sovereign.EventData{
+			Nonce: 10,
+			TransferData: &sovereign.TransferData{
+				GasLimit: 20000000,
+				Function: []byte("add"),
+				Args:     [][]byte{big.NewInt(20000000).Bytes(), make([]byte, 0)},
+			},
+		}
+
+		serialized, err := dataCodec.SerializeEventData(eventData)
+		require.Nil(t, err)
+		require.Equal(t, "000000000000000a010000000001312d00010000000361646401000000020000000401312d0000000000", hex.EncodeToString(serialized))
+	})
+}
+
+func TestDataCodec_DeserializeEventData(t *testing.T) {
```

Comment:
[comment]
Could've "compress"  this test and the one above into a single one, where you call first Serialize, then Deserialize, since you use the same consts